src/main/java/com/develhope/spring/vehicles/controller/VehicleController.java

```diff
@@ -16,12 +18,37 @@ public class VehicleController {
     @Autowired
     private VehicleService vehicleService;
 
-    @PostMapping
     public ResponseEntity<Vehicle> createVehicle(@RequestBody Vehicle vehicle) {
         Vehicle vehicleAdded = vehicleService.createVehicle(vehicle);
         return ResponseEntity.ok(vehicleAdded);
     }
 
+    @PostMapping("/shop")
+    public ResponseEntity<String> shopVehicle(@RequestBody Vehicle vehicle, @RequestParam VehicleState isPurchasable) {
+        try {
+            boolean isPurchased = vehicleService.shopVehicle(vehicle, isPurchasable);
+            if (isPurchased) {
+                return ResponseEntity.ok("Vehicle has been successfully purchased!");
```

Comment:
Non capisco se questo metodo comprende l'effettivo acquisto di un veicolo o se è solo un setter per l'enum. Purchasable indica "acquistabile", non acquistato, quindi questa richiesta post dovrebbe essere in realtà put, ed in secondo luogo andrebbe gestita utilizzando un Path Variable con l'id in cui andiamo effettivamente a prendere un veicolo già inserito in database ed impostarlo come acquistato (ripeto, ammesso che lo scopo sia quello e che non sia semplicemente un riferimento "acquistabile", e quindi nel listino).